packages/aws-rfdk/lib/deadline/test/render-queue.test.ts

```diff
@@ -291,6 +314,89 @@ describe('RenderQueue', () => {
     });
   });
 
+  describe('renderQueueSize.max', () => {
+    describe('defaults to 1', () => {
+      function assertSpecifiesMaxSize(stackToAssert: Stack) {
+        expectCDK(stackToAssert).to(haveResourceLike('AWS::AutoScaling::AutoScalingGroup', {
+          MaxSize: '1',
+        }));
+      }
+
+      test('renderQueueSize unspecified', () => {
+        // THEN
+        assertSpecifiesMaxSize(stack);
+      });
+
+      test('renderQueueSize.max unspecified', () => {
+        // GIVEN
+        const isolatedStack = new Stack(app, 'IsolatedStack');
+
+        // WHEN
+        new RenderQueue(isolatedStack, 'RenderQueue', {
+          images,
+          repository,
+          version: new VersionQuery(isolatedStack, 'Version'),
+          vpc,
+          renderQueueSize: {},
+        });
+
+        // THEN
+        assertSpecifiesMaxSize(isolatedStack);
+      });
+    });
+
+    // Deadline before 10.1.10 requires that successive API requests are serviced by a single RCS.
+    test('validates Deadline pre 10.1.10 has max value of at most 1', () => {
+      // GIVEN
+      const max = 2;
+      const newStack = new Stack(app, 'NewStack');
+      const versionOld = new VersionQuery(newStack, 'VersionOld', {version: '10.1.9'});
+      const props: RenderQueueProps = {
+        images,
+        repository,
+        version: versionOld,
+        vpc,
+        renderQueueSize: {
+          max,
+        },
+      };
+
+      // WHEN
+      expect(() => {
+        new RenderQueue(newStack, 'RenderQueue', props);
+      })
+      // THEN
+        .toThrow(`renderQueueSize.max for Deadline version less than 10.1.10.0 cannot be greater than 1 - got ${max}`);
+    });
+
+    // Asserts that when the renderQueueSize.max prop is specified, the underlying ASG's max property is set accordingly.
+    test.each([
+      [1],
+      [2],
+      [10],
+    ])('configures maximum number of ASG instances to %d', (max: number) => {
+      // GIVEN
+      const isolatedStack = new Stack(app, 'IsolatedStack');
+      const props: RenderQueueProps = {
+        images,
+        repository,
+        version: new VersionQuery(isolatedStack, 'Version'),
+        vpc,
+        renderQueueSize: {
+          max,
+        },
+      };
+
+      // WHEN
+      new RenderQueue(isolatedStack, 'RenderQueue', props);
+
+      // THEN
+      expectCDK(isolatedStack).to(haveResourceLike('AWS::AutoScaling::AutoScalingGroup', {
+        MaxSize: max.toString(),
+      }));
+    });
+  });
+
   describe('renderQueueSize.desired', () => {
```

Comment:
It looks like we're missing a test for the values of the `DesiredCapacity` of the `AWS::ECS::Service`  resource when `renderQueueSize.desired` is not passed in.

Reply:
Looks good. thanks!